src/main/java/teammates/common/util/Version.java

```diff
@@ -0,0 +1,193 @@
+package teammates.common.util;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import com.google.appengine.api.modules.ModulesService;
+import com.google.appengine.api.modules.ModulesServiceFactory;
+
+/**
+ * Represents a version by 3 parts: major version, minor version and patch version.
+ */
+public class Version {
+    private static ModulesService modulesService = ModulesServiceFactory.getModulesService();
+    private Integer major;
+    private Integer minor;
+    private String patch;
+    /**
+     * The version is represented by 3 parts: major, minor and patch.
+     * If the version has fewer than 3 numbers, the numbers will be assigned to major then to minor (if possible).
+     * Those without number will be null.
+     * 
+     * If the version has more than 3 numbers, the first number will be major, the second number 
+     * will be minor and the rest will be patch.
+     * 
+     * For example: 
+     * version = 15
+     * major = 15, minor = null and patch = null
+     * 
+     * version = 15.01
+     * major = 15, minor = 1 and patch = null
+     * 
+     * version = 15.01.03
+     * major = 15, minor = 1 and patch = "03"
+     * 
+     * version = 15.01.03.01
+     * major = 15, minor = 1 and patch = "03.01"
+     * 
+     */
+    
+    
+    /**
+     * Creates a new instance of Version from string.
+     * It accepts either XX-XX-XXXXX or XX.XX.XXXX format.
+     */
+    public Version(String versionInString) {
+        String[] list;
+        if (versionInString.contains("-")) {
+            list = versionInString.split("-", 3);   // split into at most 3 parts
+        } else {
+            list = versionInString.split(".", 3);
+        }
+        if (list.length > 0) {
+            major = Integer.parseInt(list[0]);
+        }
+        if (list.length > 1) {
+            minor = Integer.parseInt(list[1]);
+        }
+        if (list.length > 2) {
+            patch = list[2];
+        }
+    }
+    
+    /**
+     * Compares by string representation.
+     */
+    public boolean equals(Object anotherVersion) {
+        return toString().equals(anotherVersion.toString());
+    }
+    
+    /**
+     * Gets hash code for this version.
+     */
+    public int hashCode() {
+        return toString().hashCode();
+    }
+    
+    /**
+     * Converts Version to String in format XX.XX.XXXX
+     */
+    public String toString() {
+        String version = "";
+        if (major != null) {
+            version += String.valueOf(major);
+        }
+        if (minor != null) {
+            String minorInString = String.valueOf(minor);
+            if (minorInString.length() < 2) {
+                minorInString = "0" + minorInString;
```

Comment:
This means `new Version("1-1").toString()` gives `1.01`? That's a bit odd isn't it? May be you should keep the three parts as strings (in fact the entire string) and convert to numbers on demand? Alternatively, can store the original string in addition to the three numbers?